sdk/src/Core/Amazon.Runtime/Pipeline/ErrorHandler/HttpErrorResponseExceptionHandler.cs

```diff
@@ -115,6 +115,87 @@ public override bool HandleException(IExecutionContext executionContext, HttpErr
+
+            // If 404 was suppressed and successfully unmarshalled,
+            // don't rethrow the original exception.
+            if (HandleSuppressed404(executionContext, httpErrorResponse))
+                return false;
+
+            requestContext.Metrics.AddProperty(Metric.StatusCode, httpErrorResponse.StatusCode);
+
+            AmazonServiceException errorResponseException = null;
+            // Unmarshall the service error response and throw the corresponding service exception.
+            try
+            {
+                using (httpErrorResponse.ResponseBody)
+                {
+                    var unmarshaller = requestContext.Unmarshaller;
+                    var readEntireResponse = true;
+
+                    var errorContext = unmarshaller.CreateContext(httpErrorResponse,
+                        readEntireResponse,
+                        await httpErrorResponse.ResponseBody.OpenResponseAsync(),
+                        requestContext.Metrics);
+
+                    try
+                    {
+                        errorResponseException = unmarshaller.UnmarshallException(errorContext,
+                            exception, httpErrorResponse.StatusCode);
+                    }
+                    catch (Exception e)
+                    {
+                        // Rethrow Amazon service or client exceptions 
+                        if (e is AmazonServiceException ||
+                            e is AmazonClientException)
+                        {
+                            throw;
+                        }
+
+                        // Else, there was an issue with the response body, throw AmazonUnmarshallingException
+                        var requestId = httpErrorResponse.GetHeaderValue(HeaderKeys.RequestIdHeader);
+                        var body = errorContext.ResponseBody;
+                        throw new AmazonUnmarshallingException(requestId, lastKnownLocation: null, responseBody: body,
+                            innerException: e, statusCode : httpErrorResponse.StatusCode);
+                    }
+
+                    requestContext.Metrics.AddProperty(Metric.AWSRequestID, errorResponseException.RequestId);
+                    requestContext.Metrics.AddProperty(Metric.AWSErrorCode, errorResponseException.ErrorCode);
+
+                    var logResponseBody = requestContext.ClientConfig.LogResponse ||
+                        AWSConfigs.LoggingConfig.LogResponses != ResponseLoggingOption.Never;
+                    if (logResponseBody)
+                    {
+                        this.Logger.Error(errorResponseException, "Received error response: [{0}]",
+                            errorContext.ResponseBody);
+                    }
+                }
+            }
+            catch (Exception unmarshallException)
+            {
+                this.Logger.Error(unmarshallException, "Failed to unmarshall a service error response.");
```

Comment:
It has been a while, but if I recall correctly, it needed to be an exception handler because it threw on the accessor rather than giving us a way to test gracefully. I'll need to resurrect my test code to be certain.

The reason for making it a more generic exception is that, under Blazor WASM, a specific exception type was being thrown. I must confess that I was attempting to avoid creating yet another project configuration.